python/tchannel/io.py

```diff
@@ -19,8 +19,169 @@
 # THE SOFTWARE.
 
 from __future__ import absolute_import
+import os
+from .exceptions import StreamingException
+from .messages.common import StreamState
+
 
 try:
     from cStringIO import StringIO as BytesIO
 except ImportError:  # pragma: no cover
     from io import BytesIO  # noqa
+
+import tornado
+import tornado.gen
+import tornado.ioloop
+from tornado.iostream import PipeIOStream, StreamClosedError
+
+try:
+    from tornado.locks import Condition
+except ImportError:  # pragma: no cover
+    from toro import Condition
+
+from .messages import common
+
+
+class BaseArgStream(object):
+
+    def __init__(self):
+        """Base Arg Stream Class
+
+        _state is used to track the state of the arg stream.
+
+        When object is created, it is in the "init" state.
+
+        When write func is called, it will transformed to "streaming" state.
```

Comment:
I appreciate the thought behind documenting this, but implementation details like this, if documented, should use comments (`#`), not docstrings.

Things part of the object's behavior may be included in the docstring without going into implementation details. Like, "Once the stream is exhausted, it will return an empty string on reads."